File: chroma-league/src/test/java/net/booone/chromaleague/hud/parts/resource/GnarFuryBarTest.java
package net.booone.chromaleague.hud.parts.resource;

import net.booone.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class GnarFuryBarTest extends AbstractResourceTest {
    private static int currentResource;
    private static double currentRange = 400;

    @Test
    void testGnarFuryBar() {
        currentResource = 0;
        final GnarFuryBar gnarFuryBar = new GnarFuryBar();

        new AnimationTester()
                .withBeforeIterationAction(integer -> {
                    if (integer < 120) {
                        if (currentResource < 100) {
                            currentResource++;
                        }
                        currentRange = 400;
                    } else if (integer > 150) {
                        if (currentResource > 0) {
                            currentResource -= 2;
                        }
                        currentRange = 170;
                    }
                    mockResource(currentResource, 100);
                    mockRange(currentRange);
                })
                .testAnimation(gnarFuryBar, 220);
    }


}

File: test/hud/parts/resource/GnarFuryBar.test.ts
import {describe, test} from 'vitest';
import {AnimationTester} from '../../../AnimationTester.js';
import {GnarFuryBar} from '../../../../src/hud/parts/resource/GnarFuryBar.js';
import {mockRange, mockResource, setupResourceTest} from './AbstractResourceTest.js';

/** Integration test for Gnar fury resource bar on actual hardware. */
describe('GnarFury Integration', () => {
    setupResourceTest();

    test('testGnarFuryBar', async () => {
        let currentResource = 0;
        let currentRange = 400;
        const gnarFuryBar = new GnarFuryBar();

        await new AnimationTester()
            .withBeforeIterationAction(i => {
                if (i < 120) {
                    if (currentResource < 100) {
                        currentResource++;
                    }
                    currentRange = 400;
                } else if (i > 150) {
                    if (currentResource > 0) {
                        currentResource -= 2;
                    }
                    currentRange = 170;
                }
                mockResource(currentResource, 100);
                mockRange(currentRange);
            })
            .testAnimation(gnarFuryBar, 220);
    }, 30000);
});
